lib/src/rules/eol_last.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Put an end of line for non-empty files.';
+
+const _details = r'''
+
+**DO** put an end of line for non-empty files.
```

Comment:
I think you want to be clear that this is at the _end of a file_.  Maybe:

    **DO** put at least one newline at the end of non-empty files.

Reply:
Or "exactly one" if we extend it to include the full request in #1717.